### DISCUSSION TITLE
Feature Request: Support TABLESAMPLE

### DISCUSSION THREAD

**Discussion by dpxcc** [duckdb/pg_duckdb#429] [Ideas]
The `TABLESAMPLE` clause is useful for sampling from large tables, but pg_duckdb doesn't support it right now due to syntax differences in `TABLESAMPLE` clause between Postgres and DuckDB: https://duckdb.org/docs/sql/samples.html
```sql
SET duckdb.force_execution = ON;
CREATE TABLE t (a int);
SELECT * FROM t TABLESAMPLE SYSTEM (10);
```
throws
```
WARNING:  (PGDuckDB/CreatePlan) Prepared query returned an error: 'Parser Error: Sample method System cannot be used with a discrete sample count, either switch to reservoir sampling or use a sample_size
 a 
---
(0 rows)
```
but it could be an error if we added some DuckDB-only function to the query

**Comment by JelteF** [comment]
I guess this should be fairly easy to support by changing `get_tablesample_def` to include the `%` sign, which duckdb requires and postgres doesn't want. https://github.com/duckdb/pg_duckdb/blob/7e21cbe982731f559eb7d45aa543880b7de8b069/src/vendor/pg_ruleutils_17.c#L12408
